File: Angular/Hahn_Softwareentwicklung/src/app/interfaces/order.ts
import { OrderItem } from './order-item';

export interface Order {
    id: string,
    buyerId: string,
    orderDate: string,
    totalAmount: number,
    orderItems: OrderItem[],
    status: number
}

export enum OrderStatus {
    Pending = 1,
    Approved = 2,
    Rejected = 3
}

File: Backend/Entities/Order.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;

namespace Hahn_Softwareentwicklung.Entities
{
    public class Order
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        [ForeignKey("Buyer")]
        public Guid? BuyerId { get; set; }
        public DateTime OrderDate { get; set; }
        public OrderStatus Status { get; set; }
        public string PaymentMethod { get; set; }
        public decimal TotalAmount { get; set; }

        public enum OrderStatus
        {
            Pending,
            Processing,
            Shipped,
            Delivered,
            Cancelled
        }

        public Order(Guid? buyerId, decimal totalAmount, string paymentMethod)
        {
            Id = Guid.NewGuid();
            OrderDate = DateTime.Now;
            BuyerId = buyerId;
            Status = OrderStatus.Pending;
            TotalAmount = totalAmount;
            PaymentMethod = paymentMethod;
        }
    }
}
